api/src/main/java/jakarta/data/repository/Pageable.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package jakarta.data.repository;
+
+import jakarta.data.DataException;
+
+import java.util.Objects;
+import java.util.ServiceLoader;
+
+/**
+ * Abstract interface for pagination information.
+ */
+public interface Pageable {
+
+    long DEFAULT_SIZE = 10;
+    Sort EMPTY_SORT = Sort.of();
+
+    /**
+     * Returns the offset to be taken according to the underlying page and page size.
+     *
+     * @return the offset to be taken according to the underlying page and page size.
+     */
+    long getOffset();
+
+    /**
+     * Returns the page to be returned.
+     *
+     * @return the page to be returned.
+     */
+    long getPageNumber();
+
+    /**
+     * Returns the Pageable requesting the next Page.
+     *
+     * @return The next pageable.
+     */
+    Pageable next();
+
+    /**
+     * Returns the previous Pageable or the first Pageable if the current one already is the first one.
```

Comment:
We should change "or the first Pageable if the current one already is the first one" to "null if the current one already is the first one" because the previous page is never the same page and I can foresee a quirky behavior like that leading to some infinite loops in applications.